lib/util.js

```diff
@@ -147,11 +151,51 @@ function styleText(format, text, { validateStream = true, stream = process.stdou
       validateOneOf(key, 'format', ObjectKeys(inspect.colors));
     }
     if (skipColorize) continue;
-    left += escapeStyleCode(formatCodes[0]);
-    right = `${escapeStyleCode(formatCodes[1])}${right}`;
+    codes.push(formatCodes);
+  }
+
+  if (skipColorize) {
+    return text;
+  }
+
+  // Build opening codes
+  let openCodes = '';
+  for (const { 0: open } of codes) {
+    openCodes += escapeStyleCode(open);
+  }
```

Comment:
```suggestion
  for (let i = 0; i < codes.length; i++) {
    openCodes += escapeStyleCode(codes[i][0]);
  }
```
or, you'd need to use SafeArrayIterator